actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -196,7 +196,9 @@ def initialize(route, options, route_name)
             def call(t, method_name, args, inner_options, url_strategy)
               if args.size == arg_size && !inner_options && optimize_routes_generation?(t)
                 options = t.url_options.merge @options
-                options[:path] = optimized_helper(args)
+                path = optimized_helper(args)
```

Comment:
Unless I missed something, we know for sure that `path` doesn't contain a format at this point.